### PR TITLE
Initial compiled Wrapping logic.

### PR DESCRIPTION
**Pull request by vaatsalya123**
Will add more generalized cache registers for the MRUcache method.Though for now this is the initial start will explore TTL based caching as well.I used eval because exec is not a Callable function!

docs for cachetools framework: https://cachetools.readthedocs.io/en/latest/
docs for compile function in python: https://appdividend.com/2019/07/25/python-compile-example-compile-function-in-python/

### CODE REVIEW

File: ivy/func_wrapper.py [hunk 1]
```diff
@@ -40,13 +41,18 @@ def _get_first_array(*args, **kwargs):
             arr = ivy.index_nest(kwargs, arr_idxs[0])
     return arr
 
-
-# Array Handling #
+# Just a native python compile function with cache #
```
Comment (djl11) [inline ivy/func_wrapper.py:44]: Just write "Compiling with Cache", and change the number of `-` so that the title aligns correctly please :)

**Review by djl11 — CHANGES_REQUESTED** [review]

**Comment by JamieLine** [comment]
Hi!

Because Daniel has already reviewed this, am I safe to assume I don't need to look through everything unless requested? Just want to make sure people aren't being held up.

**Comment by vaatsalya123** [comment]
Hey @JamieLine yep you don't need to go through this pr Dan will go through all of it thanks.

File: ivy/func_wrapper.py [hunk 2]
```diff
@@ -40,13 +41,18 @@ def _get_first_array(*args, **kwargs):
             arr = ivy.index_nest(kwargs, arr_idxs[0])
     return arr
 
-
-# Array Handling #
+# Compiling with Cache #
```
Comment (djl11) [inline ivy/func_wrapper.py:44]: small detail but the number of `-` still don't align with the title length! aha

**Review by djl11 — CHANGES_REQUESTED** [review]
tiny change requested aha, but do we still want to merge this. Maybe we should close until we have finalised the discussion around the best design. What do you think @vaatsalya123?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**